src/nts_ke/client.rs

```diff
@@ -48,14 +48,13 @@ const DEFAULT_SCHEME: u16 = 0;
 const TIMEOUT: Duration = Duration::from_secs(15);
 
 #[derive(Clone, Debug)]
-struct ClientState {
+struct RecievedNtsKeRecordState {
     finished: bool,
-    cookies: Vec<Cookie>,
     next_protocols: Vec<u16>,
-    aead_scheme: u16,
-    next_port: u16,
-    next_server: String,
-    keys: NTSKeys,
+    aead_scheme: Vec<u16>,
+    cookies: Vec<Cookie>,
+    next_server: Option<String>,
```

Comment:
Why do you need to make `next_server` and `next_port` optional? We can just assign the default values to them if they're none.

Reply:
I think that it gets simple by using `next_server` and `next_port`  as Option type.

It is good that `deserialize()` only deserialize and `RecievedNtsKeRecordState` is just stored received NTS-KE Records is better, I think.
So, It needs to change these to Option type and be initialized by None.
The default values are now assigned by return expressions.

Reply:
If you make `KeServerConnState` looks like I mentioned, you don't need `Option` here, do you? because this struct will not exist until `deserialize` is called.

Reply:
Yes, but in nts_ke/server default value is..? Do you have any ideas?

Reply:
I choice to implement `nts_ke/server` using `process_record()`.

In NTS-KE, when the client did not send NTPv4 Server Negotiation records, `next_server` would be assumed to same to the NTS-KE server's IP address.
https://tools.ietf.org/html/draft-ietf-ntp-using-nts-for-ntp-20#section-4.1.7
In order that `nts_ke/server` decide whether the client sent this record, I make `RecievedNtsKeRecordState.next_server` to be Option type.